src/openrct2-ui/input/MouseInput.cpp

```diff
@@ -1341,7 +1341,9 @@ void InputStateWidgetPressed(
                                 dropdown_index = gDropdownDefaultIndex;
                             }
                         }
-                        WindowEventDropdownCall(cursor_w, cursor_widgetIndex, dropdown_index);
+                        WindowEventDropdownCall(
+                            cursor_w, cursor_widgetIndex,
+                            (gDropdownIsColour) ? COLOUR_UI_ORDER[dropdown_index] : dropdown_index);
```

Comment:
I don't like this its a very confusing way of doing things. It would be better if all users of colour converted their ddidx to a colour using a function call.